pkg/api/v1/orchestrator/routes/handlers.go

```diff
@@ -92,34 +92,28 @@ func (r *Routes) conditionStatusUpdate(c *gin.Context) (int, *v1types.ServerResp
 	}
 
 	// expect condition to be in an incomplete state to accept this update.
-	cr, err := r.repository.Get(ctx, serverID)
+	cond, err := r.repository.GetActiveCondition(ctx, serverID)
 	if err != nil {
-		return http.StatusInternalServerError, &v1types.ServerResponse{
-			Message: "condition lookup: " + err.Error(),
+		if errors.Is(err, store.ErrConditionNotFound) {
+			return http.StatusBadRequest, &v1types.ServerResponse{
+				Message: err.Error(),
+			}
 		}
-	}
 
-	var found bool
-	var updateable bool
-	for _, cond := range cr.Conditions {
-		if cond.Kind == conditionKind {
-			found = true
-
-			if !rctypes.StateIsComplete(cond.State) {
-				updateable = true
-			}
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: "condition lookup: " + err.Error(),
 		}
 	}
 
-	if !found {
+	if rctypes.StateIsComplete(cond.State) {
```

Comment:
Prior to this, in light of our discoveries yesterday, maybe we should check that the update is actually applicable to the active condition's ID? That is `cond.ID == conditionID`?

Reply:
added